```python
#201016_异常机制
# tryA:
#     a=int(input('请输入一个数字:'))
#     print(1/a)
# except ZeroDivisionError as e1:  #可以给异常起别名,方便打印
#     print('分母不能为0',e1)
# except ValueError as e2:  #值异常
#     print('您输入的不是数字',e2)
# except:  #捕获上面未捕获的其他未知的异常
#     print('程序出现未知错误')
# else:
#     print('没有出现错误')  #当程序正常结束时,运行一次.如果程序出现异常,不执行else语句
# finally:
#     print('程序运行完毕')  #程序结束时运行一次

# import logging  #日志模块
# import time
# import traceback  #导入异常信息显示模块
# logging.basicConfig(level='DEBUG',filename='./log1.log',filemode='a')
# logging.debug(time.strftime('%y-%m-%d %H:%M:%S')+'-------------debug')
# logging.info('--------------info')
# logging.warning('------------warning')
# logging.error('--------------error')
# logging.critical('------------critical')

# tryA:
#     a=int(input('请输入一个数字:'))
#     print(1/a)
# except ZeroDivisionError:  #可以给异常起别名,方便打印
#     logging.warning(time.strftime('%y-%m-%d %H:%M:%S')+'    '+traceback.format_exc())
# except ValueError:  #值异常
#     logging.warning(time.strftime('%y-%m-%d %H:%M:%S')+'    '+traceback.format_exc())
# except:  #捕获上面未捕获的其他未知的异常
#     print('程序出现未知错误')
# else:
#     print('没有出现错误')  #当程序正常结束时,运行一次.如果程序出现异常,不执行else语句
# finally:
#     print('程序运行完毕')  #程序结束时运行一次

#所有的异常的父类或祖类都是Exception
# NameError  #未定义的变量
# print(a)
# IndexError  #下标越界
# list1=[100,200]
# print(list1[2])
# IOError  #输入输出异常
# FileNotFoundError  #文件不存在的异常
# with open('d:/你怎么这么帅气.txt') as f1:
#     f1.read()
# print(NameError.__bases__)
# print(IndexError.__bases__)
# print(LookupError.__bases__)
# print(Exception.__bases__)
# print(BaseException.__bases__)

#使用raise关键字,手动抛出异常
# tryA:
#     raise IOError  #手动抛出异常,也就是程序会假设此处有异常
# except:
#     print('监测到IOERROR')

#断言
# assert isinstance('1.23',float)
```